File: 1993-sum-of-all-subset-xor-totals/sum-of-all-subset-xor-totals.cpp
class Solution {
public:
    int helper(vector<int>& nums,int curr,int idx) {
        if(idx == nums.size()) return curr;
        int pick = helper(nums,curr^nums[idx],idx+1);
        int unpick = helper(nums,curr,idx+1);
        return pick + unpick;
    }
    int subsetXORSum(vector<int>& nums) {
       
        return helper(nums,0,0);
       
    }
};

File: 1993-sum-of-all-subset-xor-totals/sum-of-all-subset-xor-totals.java
class Solution {
    public int Recur(int[] nums,int idx, int curr) {
        if(idx >= nums.length) return curr;
        int P = Recur(nums,idx+1,curr^nums[idx]);
        int UP = Recur(nums,idx+1,curr);
        return P+UP;
    }
    public int subsetXORSum(int[] nums) {
        return Recur(nums,0,0);
    }
}